rocketpy/rocket/aero_surface.py

```diff
@@ -1889,3 +1888,210 @@ def all_info(self):
+                the simulation always considers that at a deployment level of 0,
+                the air brakes are completely retracted and do not contribute to
+                the drag of the rocket.
+
+        reference_area : int, float
+            Reference area used to calculate the drag force of the air brakes
+            from the drag coefficient curve. Units of m^2.
+        clamp : bool, optional
+            If True, the simulation will clamp the deployment level to 0 or 1 if
+            the deployment level is out of bounds. If False, the simulation will
+            not clamp the deployment level and will instead raise a warning if
+            the deployment level is out of bounds. Default is True.
+        override_rocket_drag : bool, optional
+            If False, the air brakes drag coefficient will be added to the
+            rocket's power off drag coefficient curve. If True, during the
+            simulation, the rocket's power off drag will be ignored and the air
+            brakes drag coefficient will be used for the entire rocket instead.
+            Default is False.
+        deployment_level : float, optional
+            Current deployment level, ranging from 0 to 1. Deployment level is the
+            fraction of the total airbrake area that is Deployment. Default is 0.
+        name : str, optional
+            Name of the air brakes. Default is "AirBrakes".
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(name)
+        self.drag_coefficient_curve = drag_coefficient_curve
+        self.drag_coefficient = Function(
+            drag_coefficient_curve,
+            inputs=["Deployment Level", "Mach"],
+            outputs="Drag Coefficient",
+        )
+        self.reference_area = reference_area
+        self.clamp = clamp
+        self.override_rocket_drag = override_rocket_drag
+        self._deployment_level = deployment_level
+        self.prints = _AirBrakesPrints(self)
+        self.plots = _AirBrakesPlots(self)
+
+    @property
+    def deployment_level(self):
+        """Returns the deployment level of the air brakes."""
+        return self._deployment_level
+
+    @deployment_level.setter
+    def deployment_level(self, value):
+        # Check if deployment level is within bounds and warn user if not
+        if value < 0 or value > 1:
+            # Clamp deployment level if clamp is True
+            if self.clamp:
+                # Make sure deployment level is between 0 and 1
+                value = np.clip(value, 0, 1)
+            else:
+                # Raise warning if clamp is False
+                warnings.warn(
+                    f"Deployment level of {self.name} is smaller than 0 or "
+                    + "larger than 1. Extrapolation for the drag coefficient "
+                    + "curve will be used."
+                )
+        self._deployment_level = value
+
+    def set_deployment_level(self, deployment_level):
```

Comment:
Given the existance of the `deployment_level.setter`, is this method still necessary?

Reply:
It is there just for usability purposes. Makes it easier to make it cleared what is happening in the controller function for examples. It is technically not needed though.

Do you think we should remove it?

Reply:
I find it odd to have the property and a setter, so I would remove it.

If you really prefer keeping it, just remember that we should only change the property.setter and keep the current method only calling the property.setter.

Reply:
I am with Bressan on this one. Please refer to this past comment in this same PR. Sorry not getting into too much details.

![image](https://github.com/RocketPy-Team/RocketPy/assets/63590233/d8e6a6ce-34c1-45d5-b2de-10fd5affce27)